homeassistant/components/intesishome/climate.py

```diff
@@ -0,0 +1,407 @@
+    device_type = config[CONF_DEVICE]
+
+    controller = IntesisHome(
+        ih_user,
+        ih_pass,
+        hass.loop,
+        websession=async_get_clientsession(hass),
+        device_type=device_type,
+    )
+    try:
+        await controller.poll_status()
+    except IHAuthenticationError:
+        _LOGGER.error("Invalid username or password")
+        return
+    except IHConnectionError:
+        _LOGGER.error("Error connecting to the %s server", device_type)
+        raise PlatformNotReady
+
+    ih_devices = controller.get_devices()
+    if ih_devices:
+        async_add_entities(
+            [
+                IntesisAC(ih_device_id, device, controller)
+                for ih_device_id, device in ih_devices.items()
+            ],
+            True,
+        )
+    else:
+        _LOGGER.error(
+            "Error getting device list from %s API: %s",
+            device_type,
+            controller.error_message,
+        )
+        await controller.stop()
+
+
+class IntesisAC(ClimateDevice):
+    """Represents an Intesishome air conditioning device."""
+
+    def __init__(self, ih_device_id, ih_device, controller):
+        """Initialize the thermostat."""
+        self._controller = controller
+        self._device_id = ih_device_id
+        self._ih_device = ih_device
+        self._device_name = ih_device.get("name")
+        self._device_type = controller.device_type
+        self._connected = None
+        self._setpoint_step = 1
+        self._current_temp = None
+        self._max_temp = None
+        self._min_temp = None
+        self._target_temp = None
+        self._outdoor_temp = None
+        self._run_hours = None
+        self._rssi = None
+        self._swing = None
+        self._swing_list = None
+        self._vvane = None
+        self._hvane = None
+        self._power = False
+        self._fan_speed = None
+        self._hvac_mode = None
+        self._fan_modes = controller.get_fan_speed_list(ih_device_id)
+        self._support = SUPPORT_TARGET_TEMPERATURE | SUPPORT_FAN_MODE
+        self._swing_list = [SWING_OFF]
+
+        if ih_device.get("config_vertical_vanes"):
+            self._swing_list.append(SWING_VERTICAL)
+        if ih_device.get("config_horizontal_vanes"):
+            self._swing_list.append(SWING_HORIZONTAL)
+        if len(self._swing_list) == 3:
+            self._swing_list.append(SWING_BOTH)
+            self._support |= SUPPORT_SWING_MODE
+        elif len(self._swing_list) == 2:
+            self._support |= SUPPORT_SWING_MODE
+
+    async def async_added_to_hass(self):
+        """Subscribe to event updates."""
+        _LOGGER.debug("Added climate device with state: %s", repr(self._ih_device))
+        await self._controller.add_update_callback(self.async_update_callback)
+        try:
+            await self._controller.connect()
+        except IHConnectionError as ex:
+            _LOGGER.error("Exception connecting to IntesisHome:  %s", repr(ex))
```

Comment:
We shouldn't need to use `repr` to print the exception. The exception has a built in `__str__` magic method that by default returns the arguments of the exception. You can set default arguments by calling the constructor of the parent exception. Eg:
```py
class CustomError(Exception):
    def __init__(self):
        super().__init__(self.__class__.__name__)
```

Reply:
There's an extra space here: `:  %s`.